app/components/RuleViewerEditor/subcomponents/RuleInputOutputFieldsComponent.tsx

```diff
@@ -0,0 +1,197 @@
+    setInputOutputSchema(schemafiedInputs);
+  }, [inputOutputFields]);
+
+  const addInputField = () => {
+    updateNode(id, (draft) => {
+      if (!draft.content?.fields) {
+        draft.content = { fields: [] };
+      }
+      draft.content.fields.push({ id: crypto.randomUUID(), field: "" });
+      return draft;
+    });
+  };
+
+  const updateInputField = (item: InputOutputField, { key, label }: any) => {
+    updateNode(id, (draft) => {
+      draft.content.fields = draft.content.fields.map((input: InputOutputField) => {
+        if (input.id === item.id) {
+          input.field = key;
+          input.label = label;
+        }
+        return input;
+      });
+      return draft;
+    });
+  };
+
+  const deleteInputField = (item: InputOutputField) => {
+    updateNode(id, (draft) => {
+      draft.content.fields = draft.content.fields.filter((input: InputOutputField) => input.id !== item.id);
+      return draft;
+    });
+  };
+
+  const filterOption = (inputValue: string, option?: BaseOptionType) =>
+    (option?.label ?? "").toString().toLowerCase().includes(inputValue.toLowerCase());
+
+  const renderSelectLabel = ({ label }: { label: React.ReactNode }) => {
+    if (!label) {
+      return null;
+    }
+    const [name] = label?.toString().split(":"); // first part of label is the name (don't want description here)
+    return name;
+  };
+
+  const renderSelectOption = ({ label }: FlattenOptionData<BaseOptionType>) => {
+    if (!label) {
+      return null;
+    }
+    const [name, description] = label?.toString().split(":");
+    return (
+      <>
+        <span>{name}</span>
+        {description ? <span className={styles.selectDescription}>{description}</span> : ""}
+      </>
+    );
+  };
+
+  const renderLabel = (label: string) => {
+    const [name, description] = label?.toString().split(":");
+    return (
+      <Tooltip title={description} placement="left">
+        <span>{name}</span>
+      </Tooltip>
+    );
+  };
+
+  return (
+    <GraphNode
+      id={id}
+      specification={specification}
+      name={name}
+      isSelected={isSelected}
+      actions={
+        isEditable
+          ? [
+              <Button key="add row" type="link" onClick={addInputField} disabled={!isEditable}>
+                Add Input +
+              </Button>,
+            ]
+          : []
+      }
+    >
+      {inputOutputFields && inputOutputFields.length > 0 && (
+        <List
+          size="small"
+          dataSource={inputOutputFields}
+          renderItem={(item) => (
+            <List.Item>
+              {isEditable ? (
+                <>
+                  <Select
+                    disabled={!isEditable}
+                    showSearch
+                    placeholder="Select rule"
+                    filterOption={filterOption}
+                    options={inputOutputOptions}
+                    onChange={(value) => updateInputField(item, value)}
+                    value={item.field}
```

Comment:
Antd Select component is throwing an error for me regarding the formatting of the value on this field:
`value` should in shape of `{ value: string | number, label?: ReactNode }` when you set `labelInValue` to `true`.
updating to value={{ label: item.label, value: item.field }} removed this error for me, but unsure if this modifies intent of the field.

Reply:
Thanks for that! Updated.